shapes/neurosciencegraph/commons/list/schema.json

```diff
@@ -0,0 +1,356 @@
+      "@id": "this:IntegerNode",
+      "not": [
+        {
+          "hasValue": [
+            {
+              "@list": []
+            }
+          ]
+        }
+      ],
+      "property": [
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:rest"
+        },
+        {
+          "datatype": "xsd:integer",
+          "maxCount": 1,
+          "message": "Data should be integer type",
+          "minCount": 1,
+          "path": "rdf:first"
+        }
+      ]
+    },
+    {
+      "@id": "this:StringNode",
+      "not": [
+        {
+          "hasValue": [
+            {
+              "@list": []
+            }
+          ]
+        }
+      ],
+      "property": [
+        {
+          "datatype": "xsd:string",
+          "maxCount": 1,
+          "message": "Data should be string type",
+          "minCount": 1,
+          "path": "rdf:first"
+        },
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:rest"
+        }
+      ]
+    },
+    {
+      "@id": "this:DoubleNode",
+      "not": [
+        {
+          "hasValue": [
+            {
+              "@list": []
+            }
+          ]
+        }
+      ],
+      "property": [
+        {
+          "maxCount": 1,
+          "minCount": 1,
+          "path": "rdf:rest"
+        },
+        {
+          "datatype": "xsd:double",
+          "maxCount": 1,
+          "message": "Data should be double type",
+          "minCount": 1,
+          "path": "rdf:first"
+        }
+      ]
+    },
+    {
+      "@id": "this:ListNodeShape",
+      "@type": "NodeShape",
+      "rdfs:comment": "Defines constraints on what it means for a node to be a node within a well-formed RDF list. Note that this does not check whether the rdf:rest items are also well-formed lists as this would lead to unsupported recursion.",
+      "rdfs:label": "List node shape",
+      "or": {
+        "@list": [
+          "this:EmptyNode",
+          "this:DataNode"
+        ]
+      }
+    },
+    {
+      "@id": "this:ListOfStringsNodeShape",
+      "@type": "NodeShape",
+      "rdfs:label": "List of string node shape",
+      "message": "List should contain only strings",
+      "or": {
+        "@list": [
+          "this:EmptyNode",
+          "this:StringNode"
+        ]
+      }
+    },
+    {
+      "@id": "this:ListOfDoublesNodeShape",
+      "@type": "NodeShape",
+      "rdfs:label": "List of doubles node shape",
+      "message": "List should contain only doubles",
+      "or": {
+        "@list": [
+          "this:EmptyNode",
+          "this:DoubleNode"
+        ]
+      }
+    },
+    {
+      "@id": "this:ListOfIntegersNodeShape",
+      "@type": "NodeShape",
+      "rdfs:label": "List of integers node shape",
+      "message": "List should contain only integers",
+      "or": {
+        "@list": [
+          "this:EmptyNode",
+          "this:IntegerNode"
+        ]
+      }
+    },
+    {
+      "@id": "this:ListShape",
```

Comment:
I would suggest to keep the shsh namespace for the shapes taken from the shacl of shacl schema. shsh is already part of the https://incf.github.io/neuroshapes/contexts/schema.json:

{"@id": "shsh:ListShape"}